### PR TITLE
Add neovim nightly

### PR DESCRIPTION
**Pull request by mjlbach**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

We added support for an MSI installer in the latest neovim. I'll package stable once 0.7.0 is released next month.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/55112)

### CODE REVIEW

**Comment by mjlbach** [comment]
@justinmk @bfredl @jamessan (sorry for the ping) I used

* `Neovim.nvim.nightly` as the package identifier
* `neovim.io` as the publish`
* `Neovim` as the package name (not sure if we should use `nvim` here.
* `Apache-2.0 or Vim` as the package license

**Review by Trenly — CHANGES_REQUESTED** [review]
This PR is missing the `defaultLocale` and `version`  manifest files

**Comment by mjlbach** [comment]
Whoops! I noticed that but forgot to push, done.

**Comment by mjlbach** [comment]
wait one issue.

**Comment by mjlbach** [comment]
@Trenly should be good for review again now

File: manifests/n/Neovim/nvim/nightly/0.7.0/Neovim.nvim.nightly.installer.yaml
Comment (mjlbach) [reply manifests/n/Neovim/nvim/nightly/0.7.0/Neovim.nvim.nightly.installer.yaml:11]: That's a really good point (sorry, I'm not a windows or winget user). Wouldn't the PR need to get approved each time? I guess we could pin to a specific release instead of latest nightly.

Comment (Trenly) [reply manifests/n/Neovim/nvim/nightly/0.7.0/Neovim.nvim.nightly.installer.yaml:11]: Other than that, looks good to me. Just gotta wait for what the pipelines runs say

Comment (Trenly) [reply manifests/n/Neovim/nvim/nightly/0.7.0/Neovim.nvim.nightly.installer.yaml:11]: > That's a really good point (sorry, I'm not a windows or winget user, although this is obvious in hindsight). Wouldn't the PR need to get approved each time (so it would still get broken every day until the PR with the updated sha was merged)? I guess we could pin to a specific release instead of latest nightly, or just put this on hold until the stable release is out.

Yes, the PR would need to get merged each time. The way I've seen it done for other packages is to pin it to a specific version, and then when a new nightly build is created remove the old version and add the new version in a single PR.  #55047 is a good example of this. You don't have to remove the old version (assuming that the download link remains valid), but given that nightly builds could quickly accumulate, it's usually best to just keep the last 1-5 versions.

By doing it that way, it will ensure that the link is always valid (even if it is a few hours past the latest release), and then as soon as the new version's PR gets merged in and published, then the update seamlessly becomes available.

Comment (Trenly) [reply manifests/n/Neovim/nvim/nightly/0.7.0/Neovim.nvim.nightly.installer.yaml:11]: > Alternatively, is there an option to ignore the hash? I'm reading through the documentation/old issues but can't seem to find anything.

No, there isn't exactly an easy way to ignore the SHA. The hash is required for user safety as it is one of many ways that we validate a package hasn't changed unexpectedly. Users can bypass the check using `--force`, but only if they are not running as admin. Even then, I would still discourage bypassing the check

Comment (mjlbach) [reply manifests/n/Neovim/nvim/nightly/0.7.0/Neovim.nvim.nightly.installer.yaml:11]: I may hold off for now, we don't minor version nightly releases so it would be incompatible with our current release process (I'd have to add a dummy value 0.7.0-{1,2,3,etc.} or something in a GHA)

**Comment by sitiom** [comment]
@mjlbach

> * `Neovim.nvim.nightly` as the package identifier

This should be `Neovim.Neovim.Nightly`

> * `Neovim` as the package name (not sure if we should use `nvim` here.

That's correct, the full name should be used for the package name. `nvim` is the `Moniker`.

You might want to look at how I packaged the stable version in https://github.com/microsoft/winget-pkgs/pull/57882.

**Comment by mjlbach** [comment]
@sitiom I am a maintainer of nvim, and I  asked 2 of the other maintainers about the naming while filing this PR. Neovim is the organization, nvim is the package name. From Justin,

> I think that "neovim" as the project/org name makes sense to be distinct from nvim the text editor

FWIW I am not a Windows user so I don't personally care what the winget naming situation is.

**Comment by justinmk** [comment]
> > * `Neovim.nvim.nightly` as the package identifier
> 
> This should be `Neovim.Neovim.Nightly`
> 
> > * `Neovim` as the package name (not sure if we should use `nvim` here.
> 
> That's correct, the full name should be used for the package name. `nvim` is the `Moniker`.

I'm looking at https://github.com/microsoft/winget-cli/discussions/1672#discussioncomment-1665516 

@sitiom can you confirm that `winget install nvim` will install neovim? Neovim project would tend to choose `Neovim.Nvim.Nightly` because "neovim" is the project/marketing name. "nvim" is the text editor, CLI interface, process image name, user-facing tool, manpage name, and often the package name (except for legacy cases).

**Comment by sitiom** [comment]
> @sitiom can you confirm that `winget install nvim` will install neovim ([reference](https://github.com/microsoft/winget-cli/discussions/1672#discussioncomment-1665516))?

Yes, that works. `nvim` is set as the moniker, so that would work.

> Neovim project would tend to choose `Neovim.Nvim.Nightly` because "neovim" is the project/marketing name. "nvim" is the text editor, CLI interface, process image name, user-facing tool, manpage name, and often the package name (except for legacy cases).

`neovim` is pretty much the package name on [other packages](https://repology.org/project/neovim/packages) (except for Snap and Flatpak, which the Neovim team itself maintains); I was trying to be consistent with that.

**Comment by mjlbach** [comment]
As you pointed out we only (officially) maintain the flatpak (which uses `Neovim.nvim`) and the snap (which uses `nvim`) so it's unfortunate that the naming does not follow our conventions, but I guess it is too late now.

**Pull request closed without merging**
